pages/users/signUp/index.tsx

```diff
@@ -0,0 +1,197 @@
+    values: UserSignUpFormValues,
+    event?: React.BaseSyntheticEvent
+  ): Promise<void> {
+    event?.preventDefault();
+    if (submitting) {
+      return;
+    }
+    try {
+      action(values);
+      router.push("/users/signUp/signUp2");
+    } catch (err) {
+      setError(err.message);
+    } finally {
+      setSubmitting(false);
+    }
+  }
+
+  const togglePassword = (): void => {
+    setRevealPassword(!revealPassword);
+  };
+
+  return (
+    <StateMachineProvider>
+      <div className="form flex ml-20 mt-10 mr-32 flex-col">
+        <p className="pt-6 text-2xl h-16">Create your account</p>
+        <form className="mt-10" onSubmit={handleSubmit(onSubmit)}>
+          <fieldset>
+            <div className="flex mr-32">
+              <UserSignUpFormField
+                label="First Name"
+                name="firstName"
+                error={errors.firstName?.message}
+              >
+                <input
+                  type="text"
+                  className="input "
+                  name="firstName"
+                  placeholder="e.g., Cristiano"
+                  ref={register}
+                  defaultValue={state.userData.firstName}
+                />
+              </UserSignUpFormField>
+              <UserSignUpFormField
+                label="Last Name"
+                name="lastName"
+                error={errors.lastName?.message}
+              >
+                <input
+                  type="text"
+                  className="input "
+                  name="lastName"
+                  placeholder="e.g., Ronaldo"
+                  ref={register}
+                  defaultValue={state.userData.lastName}
+                />
+              </UserSignUpFormField>
+            </div>
+            <div className="flex mr-32">
+              <UserSignUpFormField
+                label="Email Address"
+                name="email"
+                error={errors.email?.message}
+              >
+                <input
+                  type="text"
+                  className="input "
+                  name="email"
+                  placeholder="e.g., soccer@fifa.com"
+                  ref={register}
+                  defaultValue={state.userData.email}
+                />
+              </UserSignUpFormField>
+              <UserSignUpFormField
+                label="Phone Number"
+                name="phoneNumber"
+                error={errors.phoneNumber?.message}
+              >
+                <input
+                  type="text"
+                  className="input"
+                  name="phoneNumber"
+                  placeholder="e.g., 123-456-7890"
+                  ref={register}
+                  defaultValue={state.userData.phoneNumber}
+                />
+              </UserSignUpFormField>
+            </div>
+            <UserSignUpFormField
+              label="Password"
+              name="password"
+              error={errors.password?.message}
```

Comment:
it looks like even though revealPassword is false, the form shows the password as plaintext by default - just wanted to give a heads up!

Reply:
oh shoot good catch thank you!